### PR TITLE
helm: clusterrole to add objectbucketclaims to user facing roles

### PR DESCRIPTION
**Pull request by jouve**
**Description of your changes:**

create rbac to view/edit objectbucketclaim and aggregate to [user facing roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)

view/edit/admin clusterroles are bound to a user & namespace, to manage said namespace, and are now able to create a bucket with this change. 

example for [cert-manager](https://github.com/cert-manager/cert-manager/blob/7f519b32448ace07d04f17d6b59b06d2c2643036/deploy/charts/cert-manager/templates/rbac.yaml#L403) or [elastic-operator](https://github.com/elastic/cloud-on-k8s/blob/main/deploy/eck-operator/templates/cluster-roles.yaml#L16)

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — COMMENTED** [review]
The OBCs already have rbac defined in the [rook-ceph-object-bucket clusterrole](https://github.com/rook/rook/blob/release-1.11/deploy/charts/rook-ceph/templates/clusterrole.yaml#L315-L322). Do we just need to make the labels on the clusterrole configurable? The new clusterroles seem duplicative of the existing clusterrole.

**Comment by jouve** [comment]
this clusterrole grants too many things as it is "Used for provisioning ObjectBuckets (OBs) in response to ObjectBucketClaims (OBCs) ", (ie. it is used by the operator to implement an obc into an ob+secret) 

the idea of those "user facing" clusterrole (view,edit,admin) are to be granted to a user to manage his application in a namespace (with a RoleBinding), so only namespaced resources, to consume those resources.

in "plain" words, as a user limited to a namespace, I need a bucket (obc) and attach it to my application (using the AK/SK in the secret created with the ob).

The parallel of pv/pvc/storageclass in the default "view/edit" clusterrole (kubectl get clusterrole view) :
```yaml
  apiGroups:
  - ""
  resources:
  ...
  - persistentvolumeclaims
  - persistentvolumeclaims/status
  verbs:
  - get
  - list
  - watch
``` 

with only persistentvolumeclaims and not persistentvolume or storageclass.


definition of edit/admin/view:
*  Allows admin access, intended to be granted within a namespace using a RoleBinding.
   If used in a RoleBinding, allows read/write access to most resources in a namespace, including the ability to create roles and role bindings within the namespace.
*  Allows read/write access to most objects in a namespace.
   This role does not allow viewing or modifying roles or role bindings. 
*  Allows read-only access to see most objects in a namespace. It does not allow viewing roles or role bindings.

**Comment by jouve** [comment]
regarding the Integration test CephUpgradeSuite, 
as the change in this MR is disabled by default, 
I think it's safe to say that the failure an unrelated

**Comment by BlaineEXE** [comment]
@jouve could you provide more context about the specifics of what these changes intend? I don't doubt that many of our RBAC resources could be improved a little, but I'm concerned that these changes might be specifically targeted toward users in your cluster and not something that would be helpful for all Rook users.

**Comment by jouve** [comment]
Let's assume 2 kinds of users in a cluster:
*   admin : operates the cluster, has access to everything, equivalent of root in linux
*   standard (for lack of a better word): operates applications in one (or many) namespace, (like non-root users in linux).

k8s defines roles, that it calls "user facing roles" to support this model : https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles

* cluster-admin, bound to system:masters group by default with a clusterrolebinding, it's grant access to every operation on every resource in every namespace
* admin, intended to be bound with a rolebinding (so namespaced), it allow the "standard" user to manage resources (so namespaced kind) in his namespace: pod (pod,cm,secret,*persistentvolumeclaims*, ...), apps (deploy,sts,...), etc

k8s also completes "admin" with 2 other roles:
* view: read-only role
* edit: like admin, except it can't grant role/rolebinding

To make this easier to use with custom resources, k8s does embed a controller so that clusterrole with the annotation ` rbac.authorization.k8s.io/aggregate-to-{admin/edit/view}: "true"`  are "merged" with the  admin/edit/view defined above.

that way, the admin only need to grant  admin/edit/view to "standard" users once, and fix admin/edit/view roles for new custom resources (using the mechanism described above).

=> this model is provided by k8s, not "my" model.


objectbucketclaim being the object storage equivalent of pvc (pvc is in the admin/edit/view) , I think it's safe to add have the feature to add it to the 3 roles admin/view/edit (disabled by default in this PR for backward compatibility).

for example: a "standard" (with role admin in his namespace) user deploy [nextcloud](https://nextcloud.com/fr/) (a file hosting app, like google drive) in his namespace, nextclound can use filesystem storage or object storage as a backend:
* the user creates a pvc, mount it in the pod, configure the app for local storage
* the user creates a obc, inject the secret in the pod, configure the app for object storage

**Comment by parth-gr** [comment]
@jouve any updates

**Comment by jouve** [comment]
thanks for reopening :)

what would be needed to progress ?

**Comment by parth-gr** [comment]
Is there changes needed to add or you are waiting for the reviews

**Comment by jouve** [comment]
I replied to @travisn  and @BlaineEXE. Idk if they were convinced ?!

File: deploy/charts/rook-ceph/values.yaml
Comment (travisn) [inline deploy/charts/rook-ceph/values.yaml:60]: The roles are currently very specific to OBCs. Perhaps this setting should be more specific to OBCs, and if any other RBAC is aggregated in the future, they would have separate settings. What about this?
```
rbacAggregate:
  enableOBCs: true
```

File: deploy/charts/rook-ceph/templates/aggregate-roles.yaml
Comment (BlaineEXE) [inline deploy/charts/rook-ceph/templates/aggregate-roles.yaml:2]: Are these roles automatically added to any k8s user in the view/edit group, or are they only enabled for users after more admin control? I wonder if it could be easier to include these as default roles created by Rook, which would not add complexity to the helm chart. As long as doing so wouldn't accidentally give users extra permissions, maybe we should consider that.

Comment (jouve) [reply deploy/charts/rook-ceph/templates/aggregate-roles.yaml:2]: the roles are added to the [cluster roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)  view / edit / admin respectively  thanks to the [ClusterRole aggregation](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles).

view/edit/admin are default roles of the ClusterRole aggregator.

then a "human" admin binds those role to users in a namespace (using rolebinding), so it's disabled by default to ensure backward compatibility.

File: deploy/charts/rook-ceph/values.yaml
Comment (jouve) [reply deploy/charts/rook-ceph/values.yaml]: done

**Review by travisn — CHANGES_REQUESTED** [review]
Please squash the commits and also add a description paragraph as described in the [commit structure](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure). Otherwise, LGTM

**Review by travisn — APPROVED** [review]

**Pull request merged**
